fiftyone/utils/eval/coco.py

```diff
@@ -89,39 +111,295 @@ def evaluate_image(
+                        if m[4]:
+                            continue
+
+                        c = m[0] if m[0] != None else m[1]
+                        if c not in classes:
+                            classes.append(c)
+
+                        if c not in thresh_matches[t]:
+                            thresh_matches[t][c] = {
+                                "tp": [],
+                                "fp": [],
+                                "num_gt": 0,
+                            }
+
+                        if m[0] == m[1]:
+                            thresh_matches[t][c]["tp"].append(m)
+                        elif m[1]:
+                            thresh_matches[t][c]["fp"].append(m)
+
+                        if m[0]:
+                            thresh_matches[t][c]["num_gt"] += 1
+
+        # Compute precision-recall array
+        # Reference:
+        # https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/cocoeval.py
+        precision = -np.ones((len(iou_threshs), len(classes), 101))
+        recall = np.linspace(0, 1, 101)
+        for t in thresh_matches.keys():
+            for c in thresh_matches[t].keys():
+                tp = thresh_matches[t][c]["tp"]
+                fp = thresh_matches[t][c]["fp"]
+                num_gt = thresh_matches[t][c]["num_gt"]
+                if num_gt == 0:
+                    continue
+
+                tp_fp = [1] * len(tp) + [0] * len(fp)
+                confs = [p[3] for p in tp] + [p[3] for p in fp]
+                inds = np.argsort(-np.array(confs), kind="mergesort")
+                tp_fp = np.array(tp_fp)[inds]
+                tp_sum = np.cumsum(tp_fp).astype(dtype=np.float)
+                total = np.arange(1, len(tp_fp) + 1).astype(dtype=np.float)
+
+                pre = tp_sum / total
+                rec = tp_sum / num_gt
+
+                q = np.zeros((101,))
+                for i in range(len(pre) - 1, 0, -1):
+                    if pre[i] > pre[i - 1]:
+                        pre[i - 1] = pre[i]
+
+                inds = np.searchsorted(rec, recall, side="left")
+                try:
+                    for ri, pi in enumerate(inds):
+                        q[ri] = pre[pi]
+                except:
+                    pass
+
+                precision[iou_threshs.index(t)][classes.index(c)] = q
+
+        return COCODetectionResults(
+            matches, precision, recall, iou_threshs, classes, missing=missing
+        )
+
+
+class COCODetectionResults(DetectionResults):
+    """Class that stores the results of a COCO detection evaluation.
+
+    Args:
+        matches: a list of ``(gt_label, pred_label, iou, pred_confidence)``
+            matches. Either label can be ``None`` to indicate an unmatched
+            object
+        precision: an array of precision values of shape
+            ``num_iou_threshs x num_classes x num_recall``
+        recall: an array of recall values
+        iou_threshs: the list of IoU thresholds
```

Comment:
I passed `iou_threshs` through here as well since the `precision` array's size depends on it (so this would be interesting information to the user to access via `results.iou_threshs`